### PR TITLE
Eliayazdi-homework-w1

### PR DESCRIPTION
**Pull request by EliaYazdi**

### CODE REVIEW

File: week1/homework/src/server.js [hunk 1]
```diff
@@ -1,20 +1,49 @@
...
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
- */
 function createServer(port) {
   let state = 10;
 
-  const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+  const server = http.createServer((req, response) => {
+    function printData() {
```
Comment (marianabocoi) [inline week1/homework/src/server.js]: Great job!
Try moving the printData function outside the  createServer function

File: week1/homework/src/server.js [hunk 2]
```diff
@@ -1,20 +1,49 @@
...
+      response.writeHead(200, {
+        'Content-Type': 'application/json'
+      });
+      let json = JSON.stringify({
```
Comment (marianabocoi) [inline week1/homework/src/server.js]: you can instead of defining the variable separately use it in the function call 

This is called to inline the variable (see what they say inlining is here: https://blog.jetbrains.com/webstorm/2009/10/javascript-refactoring-inline-variable )

File: week1/homework/src/server.js [hunk 3]
```diff
@@ -1,20 +1,49 @@
...
+      printData();
+    }
+    else if (req.url == '/reset') {
+      state = 10;
+      printData();
+    }
+    else {
+      response.writeHead(404, {
+        'Content-Type': 'application/json'});
+      let json = JSON.stringify({
```
Comment (marianabocoi) [inline week1/homework/src/server.js]: inline (see the previous comment) this variable as well.

File: week1/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (marianabocoi) [inline week1/homework/src/index.js:1]: please revert this change to the index file 
(delete newline)

**Review by marianabocoi — COMMENTED** [review]
Good job! 
Minor comments

File: week1/homework/src/server.js (hunk 1 above)
Comment (EliaYazdi) [reply week1/homework/src/server.js]: Done!

File: week1/homework/src/server.js (hunk 2 above)
Comment (EliaYazdi) [reply week1/homework/src/server.js]: I think I got your point.

File: week1/homework/src/server.js [hunk 5]
```diff
@@ -1,20 +1,48 @@
+
 'use strict';
 
 const http = require('http');
 
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
- */
+function printData(state, response) {
```
Comment (apooslr) [inline week1/homework/src/server.js:6]: you dont have to change this, but good thing  to keep in mind is that naming is important, and giving accurate and descriptive names  for functions and variables is good practice :)

File: week1/homework/src/server.js [hunk 6]
```diff
@@ -1,20 +1,48 @@
...
+      state -= 1;
+      printData(state, response);
+    }
+    else if (req.url == '/reset') {
+      state = 10;
+      printData(state, response);
+    }
+    else {
+      response.writeHead(404, {
+        'Content-Type': 'application/json'});
```
Comment (apooslr) [inline week1/homework/src/server.js:37]: This is the same thing you do on line 7 och  8, maybe you could move this in to you `printData` function ?

**Review by marianabocoi — APPROVED** [review]

**Review by marianabocoi — APPROVED** [review]
👍
